inbox/api/ns_api.py

```diff
@@ -532,6 +533,14 @@ def folders_labels_create_api():
     valid_display_name(g.namespace.id, category_type, display_name,
                        g.db_session)
 
+    if g.namespace.account.provider not in ['gmail', 'eas']:
+        # Translate the name of the folder to an actual IMAP name
+        # (e.g: "Accounting/Taxes" becomes "Accounting.Taxes")
+        display_name = imap_folder_path(
```

Comment:
Hmm so whatever the behavior is, this diff doesn't change it, just moves the call earlier into the initial save instead of a subsequent save. 

It looks like `imap_folder_path` checks that the path does not already start with the prefix though, so we should be good. (https://github.com/nylas/sync-engine/blob/master/inbox/util/misc.py#L220)